tutorial.md

```diff
@@ -0,0 +1,165 @@
+* [kn](https://knative.dev/docs/install/install-kn/) CLI
+
+
+## Cluster Setup
+To use Boson Functions, you'll need a Kubernetes cluster with Knative Serving
+and Eventing installed. If you have a recent version of OpenShift, you can
+simply install the Serverless Operator. If you don't have a cluster already,
+you can create a simple cluster with [kind](https://kind.sigs.k8s.io/). Follow
+these [step by step instructions](kind-setup.md) to install on your local
+machine.
+
+## Boson Tooling
+The primary interface for Boson project is the `faas` CLI.
+[Download][faas-download] the most recent version and install it some place
+within your `$PATH`.
+
+```sh
+# Be sure to download the correct binary for your operating system
+curl -L -o - faas.gz https://github.com/boson-project/faas/releases/download/v0.6.0/faas_linux_amd64.gz | gunzip > faas && chmod 755 faas
+sudo mv faas /usr/local/bin
+```
+## Configuring a Container Repository
+The unit of deployment in Boson Functions is an [OCI](https://opencontainers.org/)
+container image, typically referred to as a Docker container image.
+
+In order for the `faas` CLI to manage these containers, you'll need to be
+logged in to a container registry.
+
+
+```bash
+docker login -u <username> -p <password> <repository.url>
+```
+
+> Note: many of the CLI commands take a `--repository` argument. Set the `FAAS_REPOSITORY` environment variable in order to omit this parameter when using the CLI.
+
+```bash
+export FAAS_REPOSITORY=docker.io/<repo.name>
+```
+
+## Creating a Project
+With your Knative enabled cluster up and running, you can now create a new
+Function Project. Let's start by creating a project directory. Function names
+in `faas` correspond to URLs at the moment, and there are some finicky cases
+at the moment. To ensure that everything works as it should, create a project
+directory consisting of three URL parts. Here is a good one.
+
+```bash
+mkdir fn.example.io
+cd fn.example.io
+```
+
+Now, with one command we will create the project files, build a container, and
+deploy the function as a Knative service.
+
+
+```bash
+faas create -l node -y
+```
+
+This will create a Node.js Function project in the current directory accepting
+all of the defaults inferred from your environment, for example`$FAAS_REPOSITORY`.
+When the command has completed, you can see the deployed function.
+
+```bash
+$ kn service list
+NAME            URL                                          LATEST                  AGE   CONDITIONS   READY   REASON
+fn-example-io   http://fn-example-io.faas.127.0.0.1.nip.io   fn-example-io-ngswh-1   24s   3 OK / 3     True
+```
+
+Clicking on the URL will take you to the running function in your cluster.
+
+## Local Development
+The `faas create` command also results in a docker container that can be run locally
+with container ports mapped to localhost.
+
+```bash
+faas run
```

Comment:
running this command asks me to install appsody:

```
Error: please install 'appsody'
```

Reply:
Try downloading [`v0.6.1`](https://github.com/boson-project/faas/releases/tag/v0.6.1). That should not happen in the latest.

Reply:
success!